investment-app/TRINV.Api/TRINV.Application/Interfaces/IInvestmentAppDbContext.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace TRINV.Application.Interfaces
+{
+
+    public interface IInvestmentAppDbContext
```

Comment:
@glachezar  Do we really need a interface for the Context? 
What are we gonna addhere ?

Reply:
I've done some research on the matter. I have to say I was wrong. As you probably remember, I told you that the IDbContext interface is needed to retrieve data from the database. Well, that's true, but it turns out it's not the only way.

The other method involves using a specific repository pattern for each entity, with interfaces defined in the domain layer. Additionally, this approach requires using UnitOfWork, and the IUnitOfWork interface has to be defined in the application layer.

The first approach using the IDbContext interface offers faster implementation, but the downside is that if we need to switch the database type (for example, from relational to non-relational), all the handlers in the application layer would need to be rewritten.

The second approach takes longer to implement, but it means we won't be solely dependent on EfCore within the application layer. It's easier for testing and mocking. Plus, if we ever switch between relational and non-relational databases, only the repository implementations need to be rewritten.